src/gf_layers_layer_util/include/gf_layers_layer_util/util.h

```diff
@@ -112,7 +113,11 @@ class ProtectedMap {
   // Warning: we currently require the values to have pointer stability.
   // This is guaranteed for std::unordered_map.
   using InternalMapType = MapTemplate<KeyType, ValueType>;
-  using ThreadLocalCacheType = std::pair<KeyType, ValueType*>;
+
+  size_t count(KeyType key) const {
+    ScopedLock lock(mutex_);
+    return map_.count(key);
+  }
```

Comment:
I just realised: we should probably not have a count function.

Use `get` if you want to maybe get something from the map.

Use `put` if you want to maybe put something in the map; check the bool return value to see if you actually did put it in the map.

Reply:
Yeah, I realised the same. The way I was using count() was to check if the map contained the given key, so I replaced `count(key)` with `contains(key)` that is just a shortcut and more readable form of `get(key) != nullptr`. 

And btw, functions in this file don't follow the function naming rules.

Reply:
Yeah my other concern is: `count` or `contains` encourages you to check something and then act upon the result, without holding the mutex throughout. It doesn't really matter in most cases, but I would prefer to not have `count` or `contains`. `get` and `put` kind of force you to attempt your action atomically.

Good point about the naming. I am not sure when to update these. Perhaps I should it now and you can rebase, as I have already made other changes, and you will probably need to rebase/merge anyway.

Reply:
I see, that makes sense.

Reply:
Oh, and go ahead with the naming updates.

Reply:
I think I will wait for naming updates, as I made some changes to util.h in this PR. Once we merge, I can quickly rename everything. For now, feel free to follow the existing style.